scripts/india_census/primary_religion_data/preprocess.py

```diff
@@ -147,82 +215,112 @@ def stat_var_index_key(row, data_category_column=None):
             name_array.append("ScheduledTribe")
+            constraints_array.append("socialCategory: dcs:ScheduledTribe")
 
         if row["literacyStatus"] == "Literate":
             name_array.append("Literate")
-        if row["literacyStatus"] == "Illiterate":
+            constraints_array.append("literacyStatus: dcs:Literate")
+        elif row["literacyStatus"] == "Illiterate":
             name_array.append("Illiterate")
-        else:
-            pass
+            constraints_array.append("literacyStatus: dcs:Illiterate")
 
         if row["workerStatus"] == "Worker":
             if row["workerClassification"] == "MainWorker":
                 name_array.append("MainWorker")
+                constraints_array.append("workerClassification: dcs:MainWorker")
                 if row["workCategory"] != "":
                     name_array.append(row["workCategory"])
+                    constraints_array.append("workCategory: dcs:" +
+                                             row["workCategory"])
 
             elif row["workerClassification"] == "MarginalWorker":
                 name_array.append("MarginalWorker")
+                constraints_array.append(
+                    "workerClassification: dcs:MarginalWorker")
 
                 if row["workCategory"] != "":
                     name_array.append(row["workCategory"])
+                    constraints_array.append("workCategory: dcs:" +
+                                             row["workCategory"])
 
                 if row["workPeriod"] == "[Month - 3]":
                     name_array.append("WorkedUpto3Months")
+                    constraints_array.append("workPeriod:" + row["workPeriod"])
 
                 if row["workPeriod"] == "[Month 3 6]":
                     name_array.append("Worked3To6Months")
+                    constraints_array.append("workPeriod:" + row["workPeriod"])
+
             else:
                 name_array.append("Workers")
 
         elif row["workerStatus"] == "NonWorker":
             name_array.append("NonWorker")
+            constraints_array.append("workerStatus: dcs:NonWorker")
 
         if place_of_residence == "Urban":
             name_array.append("Urban")
+            constraints_array.append(
+                "placeOfResidenceClassification: dcs:Urban")
             row["description"] = row["description"] + " - Urban"
 
         elif place_of_residence == "Rural":
             name_array.append("Rural")
+            constraints_array.append(
+                "placeOfResidenceClassification: dcs:Rural")
             row["description"] = row["description"] + " - Rural"
 
         if row["gender"] == "Male":
             name_array.append("Male")
+            constraints_array.append("gender: schema:Male")
 
         elif row["gender"] == "Female":
             name_array.append("Female")
+            constraints_array.append("gender: schema:Female")
 
         name = "_".join(name_array)
```

Comment:
name components are typically sorted alphabetically by property, so 'Work*' may have to go to the end.

You can also use data/util/statvar_dcid_generator.py:get_statvar_dcid although it needs a constraints dict rather than an array. That also handles quantity ranges and prefixes for Work*

Reply:
Will do the sorting.

For now I keep this. In the next round. I will use `statvar_dcid_generator`

Reply:
@ajaits  last year wehn we imported the data, for example [this](Node: dcid:Count_Person_ScheduledCaste_MarginalWorker_Urban_Male). The order was something like his
```
Node: dcid:Count_Person_ScheduledCaste_MarginalWorker_Urban_Male
description: "Male Marginal Workers - Urban"
typeOf: dcs:StatisticalVariable
populationType: dcs:Person
statType: dcs:measuredValue
measuredProperty: dcs:count
socialCategory: dcs:ScheduledCaste
workerStatus: dcs:Worker
workerClassification: dcs:MarginalWorker
placeOfResidenceClassification: dcs:Urban
gender: schema:Male
```

I have kept the same here. Should I change for this PR alone?

Reply:
@pradh let me know I will change.

Reply:
I don't have a strong opinion one way or the other, but if we want to change it, better to do it in this PR.

As for the existing ones, better to let them be, for the risk of changing existing SV ID that might be used in URLs etc.

Reply:
@ajaits can we keep this as it is. It will be in sync with the rest of the census imports done.